tests/unit_tests/test_wave_06.py

```diff
@@ -261,11 +279,58 @@ def test_swap_best_by_category_no_other_match_is_false():
         their_priority="Decor"
     )
 
-    raise Exception("Complete this test according to comments below.")
+    
+    assert not result 
+    assert len(tai.inventory) == 3
+    assert len(jesse.inventory) == 3
+    assert item_a and item_b and item_c in  tai.inventory 
+    assert item_d and item_e and item_f in jesse.inventory
+
+
     # *********************************************************************
     # ****** Complete Assert Portion of this test **********
     # *********************************************************************
     # Assertions should check:
     # - That result is falsy
     # - That tai and jesse's inventories are the correct length
     # - That all the correct items are in tai and jesse's inventories
+
+def test_swap_newest_item_returns_True():
+
+    #Arrange 
+    item_a = Decor(condition=2.0, age= 1)
+    item_b = Electronics(condition=4.0, age= 2)
+    item_c = Decor(condition=4.0, age= 3)
+    tai = Vendor(
+        inventory=[item_c, item_b, item_a]
+    )
+
+    item_d = Clothing(condition=2.0, age= 0)
+    item_e = Decor(condition=4.0, age=3)
+    item_f = Clothing(condition=4.0, age=4)
+    jesse = Vendor(
+        inventory=[item_f, item_e, item_d]
+    )    
+
+    #Act 
+    result = tai.swap_by_newest(jesse)
+
+    #Assert
+    assert result 
+    assert len(tai.inventory) == 3
+    assert len(jesse.inventory) == 3
+    assert item_d and item_b and item_c in  tai.inventory 
```

Comment:
The `and` usage made it so you don't test what you wanted to test here. `item_d` should have swapped...but if it did *not* swap, your test would not fail! To see that in action, comment out the test's Act statement and the `assert result` on line 319. The assertions pass even if no swap occurs!